Moved Table related components to common component file of Table

MVarshini opened the pull request.

MVarshini committed:
Corrections from running the dashboard

Contoller name check and Date picker

changed fetchControllers method

removed return in fetchDataList

mainlayout changes

shubham-html-css-js commented on dashboard/src/modules/components/TableComponent/index.jsx:
```diff
@@ -84,8 +92,13 @@ export const TableWithFavorite = () => {
+        repo.name === element.name &&
+        repo.controller === element.controller &&
+        repo.metadata["dataset.created"] === element.metadata["dataset.created"]
+      )
```
comparing only names will get the job done,since name of each and every dataset will be unique

MVarshini replied:
Changed `for` loop to `for-of` loop. Functionality remains the same

MVarshini replied:
Modified the code to check only for the controller name

shubham-html-css-js commented on dashboard/src/modules/components/TableComponent/common-components.jsx:
```diff
@@ -0,0 +1,92 @@
+            new Date((data.metadata["dataset.created"]).split(":")[0])>= startDate &&
+            new Date((data.metadata["dataset.created"]).split(":")[0])<= new Date(endDate)
+        );
+      });
```
There is a filter Date utility function in utils folder.We can use that here

MVarshini replied:
This is the function in SearchComponent. I have moved it to common-components file. Please do the necessary changes as needed.

dbutenhof committed:
Remove controllers/months API

PBENCH-639

The range of months (YYYY-MM) containing Elasticsearch run documents for
a specific controller name is no longer of any interest, so remove this
obsolete API.

dbutenhof committed:
Remove "indices" section from endpoints API response (#2808)

PBENCH-642

Now that we have a complete "V1" API, there's no further need for the
server to help clients access the server's Elasticsearch indices
directly; it's time to remove the index information from the endpoints
configuration response.

dbutenhof committed:
Remove controllers/list API (#2807)

PBENCH-640

The Elastic-based controllers/list API is obsolete as we move to focus
on USERS and DATASETS instead of relying on the old concept of the
"controller" (which was a standin for USER when testers could rely on
identifying their own dedicated systems).

siddardh-ra committed:
- Move abort() call to top-level API handling (#2730)

PBENCH-202

Move all abort() calls to top-level API handling in ApiBase._dispatch() to be sure that Flask's
abort exception isn't inadvertently converted to an INTERNAL_SERVER_ERROR via an
except Exception clause.

portante committed:
Correct spelling in `pidstat` tool help text

portante committed:
Fix Tool Meister instance exit code handling

From testing we noticed that the `start()` method was not properly
returning the exit code from the `driver()` method.  We also remove
the gaps in the error codes returned.

dbutenhof committed:
Convert server APIs to ISO 8601 UTC dates (#2804)

* Convert server APIs to ISO 8601 UTC dates

PBENCH-638

SQLAlchemy stores SQL dates in "naive UTC" by default; add logic to make
this explicit and to add adjustment of aware dates on input and creation
of aware UTC dates on retrieval.

Adopt ISO 8601 formatting for dates reported through the API. (This does
not include a rigorous analysis of Elasticsearch data, although at least
some already appear in ISO 8601 UTC.)

riya-17 committed:
Add missing `f` letter for f-string

dbutenhof committed:
Fix "index map" APIs to use dataset name (#2812)

* Fix "index map" APIs to use dataset name

PBENCH-644

To align with the service layer of the old dashboard, the search,
mappings, and contents APIs were designed to identify datasets using
the Elasticsearch run document MD5 (run_id or run.id) rather than the
more straightforward model of relying on the dataset name.

As we approach the verge of using TOC data in the new dashboard, fixing
this seems timely.

portante committed:
Isolate parameter transforms to ToolDataSink class

We streamline parameter passing and front-load parameter object
creation.

We also remove the use of the `bash`-like uppercase variable, `PROG`, replacing it with the Path object `name`, and/or, renaming the variable to `logger_name`.  We also rename the ToolGroup `group` field name to `name`, and use that instead of a variable, along with changing `group` variable names to `tool_group` to help with code readability.

Co-authored-by: Webb Scales <wscales@redhat.com>

dbutenhof committed:
Convert authorization to use ID rather than string (#2825)

PBENCH-646

User authorization via ApiBase._check_authorization has always used
the "raw" specified username string from the client. It makes more sense
to use the validated internal form (the stringified DB row ID), and this
in particular will simplify handling of a username that appears in a URL
query parameter.

dbutenhof committed:
New API to fetch date range of accessible datasets (#2820)

* New API to fetch date range of accessible datasets

PBENCH-643

Add a new `datasets/daterange` API to return the earliest and
latest creation date of datasets readable by the authenticated
user and with specified owner and access filters.

dbutenhof committed:
Fix GET /datasets/list authorization (#2827)

Fix GET /datasets/list authorization

PBENCH-648

When I deployed my new server, I found that doing an unauthorized
/datasets/list query was failing with 500. This is because the
`UnauthorizedAccess` exception is handled only in `_dispatch` where we
authorized automatically based on a JSON body payload.

My original "vision" had been to make both `UnauthorizedAccess` and
`SchemaError` extend `APIAbort` to simplify the logic, but I lost track
of that during the review. This PR fixes that, and streamlines the error
logic a bit.

webbnh committed:
Quick v0.71 fixes (#2821)

Quick fixes for v0.71; pick lint

ldoktor committed:
Fix pbench-fio local fio-server execution

With the separation of local vs. remote hosts handling in pbench-fio
(0dabb2392cacb46103fed7757426bc49319b18f5) the local execution was
modified without double quotes, which results in variables not being
filled.

Similarly the path that used to be set by the ssh command is not set so
the client-result.* files are being placed into the current directory.
Let's explicitly set the output path to the usual location.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>
Co-authored-by: Peter Portante <peter.portante@redhat.com>

shubham-html-css-js committed:
Added the browsing page for non-logged in users (#2733)

Add react components and infrastructure to support unauthenticated users browsing public datasets.

dbutenhof committed:
Add authentication to /datasets/metadata (#2828)

* Add authentication to /datasets/metadata

PBENCH-649

When I fixed the exception propagation for /datasets/list, I noticed
that /datasets/metadata wasn't authorizing access at all, either for
GET or PUT. This adds (and tests) authorization checks, which requires a
new mechanism in `_check_authorization` to override the class default
API role.

portante committed:
Add UUIDs to command line of running Tool Meisters

In order to facilitate a "big-hammer" approach to cleaning up wayward
Tool Meister sub-systems, we add UUIDs to the command line invocations
of the remote Tool Meister processes.  We don't need to do that for the
local Redis server, Tool Data Sink, and Tool Meister process because we
have their PID files already.

We do take this opportunity, though, to verify the UUID given on the
command line against the running parameters stored in Redis.

npalaska committed:
Remove minor redundancy in `dataset_mappings`

Remove bit of redundancy in `dataset_mappings` for getting ES mapping
properties.

dbutenhof committed:
Move towards embedded URL parameters: metadata

PBENCH-669

We started out with all parameters in a JSON request payload to `POST`
operations following Elasticsearch conventions. While some new APIs have
used both query parameters and embedded Flask URL parameters, there are
older APIs that should be retrofitted to that model.

In this PR, I'm addressing `/datasets/metadata` (which seems like a good
place to start as the Overview page is on the cusp of using it).

In particular, we'll now `GET`|`PUT`
`http://<server>/api/v1/datasets/metadata/<name>`. rather than
specifying the dataset name with `?name=<name>`.

MVarshini committed:
updated Search Component

dbutenhof committed:
New user metadata namespace for Overview page (#2826)

* New user metadata namespace for Overview page

PBENCH-645

This change repurposes the "user.*" Dataset Metadata namespace as a per-user
key space, where each authenticated client with READ access to a given dataset
will be able to write independent metadata attached to that dataset. (Only
the owner has UPDATE access to change the "global" metadata keys in the
"dashboard.*" namespace.)

At the Metadata internal class level, methods now allow specifying a
User to associate each item of metadata. At the API level, the user is
always the authenticated user (or None for only global metadata).

(An unauthenticated client can ask for `user.*` keys, but the value will
always be None as authenticated clients will never create global values
for the `user.*` namespace and unauthenticated clients can't ever set
metadata values.

MVarshini committed:
replaced the term controllers

MVarshini committed:
Merge branch 'main' into review_pr_2733
